api/pkg/observability/deployment/identity.go

```diff
@@ -0,0 +1,52 @@
+package deployment
+
+import (
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+)
+
+func enrichIdentityToPod(podSpec corev1.PodSpec, secretName string, containerNames []string) corev1.PodSpec {
+	secretVolume := createVolumeFromSecret(secretName)
+	updatedPodSpec := podSpec.DeepCopy()
+
+	containerExist := false
+	for idx, containerSpec := range updatedPodSpec.Containers {
+		updatedContainerSpec := containerSpec
+		for _, targetName := range containerNames {
+			if targetName != updatedContainerSpec.Name {
+				continue
+			}
```

Comment:
nit: We probably aren't expecting the number of elements in `containerNames` to be super long, but can we make `containerNames` into something like a set (e.g. `map[string]struct{}`, or the set type from the [golang-collections](https://pkg.go.dev/github.com/golang-collections/collections/set) package, etc.) to check for the presence of containers in it instead of looping through a list? It feels sad to be looping through nested loops 😥